database/scrapers/organizations.ts

```diff
@@ -0,0 +1,249 @@
+      const links = [];
+      for (const linkId of org.links) {
+        try {
+          const resp = (await this.fetchJSON(
+            `${this.urls.links}/${linkId}`,
+            `link ${linkId}`,
+          )) as DirectusSingleResponse<DirectusLink>;
+          const link = resp.data;
+          links.push({
+            id: link.id,
+            link: link.link,
+            type: this.detectLinkType(link.link),
+          } as StudentOrganizationLink);
+        } catch (e) {
+          this.logger.warning(`Failed to fetch link ${linkId}: ${e}`);
+        }
+      }
+      await orgModel.related("links").createMany(links);
+    }
+  }
+
+  private detectLinkType(link: string): LinkType {
+    let url;
+    try {
+      url = new URL(link);
+    } catch {
+      this.logger.warning(
+        `Failed to parse social link '${link}' - assigning the Default linktype`,
+      );
+      return LinkType.Default;
+    }
+    if (url.protocol === "mailto:") {
+      return LinkType.Mail;
+    }
+    if (url.protocol !== "https:" && url.protocol !== "http:") {
+      this.logger.warning(
+        `Encountered an unknown protocol '${url.protocol}' in social link '${link}' - assigning the Default linktype`,
+      );
+      return LinkType.Default;
+    }
+    for (const [domain, type] of this.linkDomains) {
+      if (url.host === domain || url.host.endsWith(`.${domain}`)) {
+        return type;
+      }
+    }
+    this.logger.info(
+      `Social link '${link}' matched no domains - assigning the Default linktype`,
+    );
+    return LinkType.Default;
+  }
+
+  private convertSource(source: string): OrganizationSource {
+    switch (source) {
+      case "student_department":
+        return OrganizationSource.StudentDepartment;
+      case "aktywni_website":
+        return OrganizationSource.PwrActive;
+      case "manual_entry":
+        return OrganizationSource.Manual;
+    }
+    this.logger.warning(
+      `Unknown source '${source}' - assigning the Manual source`,
+    );
+    return OrganizationSource.Manual;
+  }
+
+  private convertType(type: string): OrganizationType {
+    switch (type) {
+      case "scientific_cirlce": //XDDDDDD
+        return OrganizationType.ScientificCircle;
+      case "student_organization":
+        return OrganizationType.StudentOrganization;
+      case "student_media":
+        return OrganizationType.StudentMedium;
+      case "cultural_agenda":
+        return OrganizationType.CultureAgenda;
+      case "student_council":
+        return OrganizationType.StudentCouncil;
+    }
+    this.logger.warning(
+      `Unknown type '${type}' - assigning the Student Organization type`,
+    );
+    return OrganizationType.StudentOrganization;
+  }
+  private async newAsset(directusId: string): Promise<string> {
+    let extension = (
+      (await this.fetchJSON(
+        `https://admin.topwr.solvro.pl/files/${directusId}?fields=filename_disk`,
+        directusId,
+      )) as FileMetaResponse
```

Comment:
make the item name param more precise here: something like `file metadata for '${directusId}'` would be better.